pkg/controller/acmeorders/controller.go

```diff
@@ -62,70 +71,94 @@ type controller struct {
 	// so the handleOwnedResource method can enqueue resources
 	queue workqueue.RateLimitingInterface
 
+	// scheduledWorkQueue holds items to be re-queued after a period of time.
+	scheduledWorkQueue scheduler.ScheduledWorkQueue
+
 	// logger to be used by this controller
 	log logr.Logger
 }
 
-// Register registers and constructs the controller using the provided context.
-// It returns the workqueue to be used to enqueue items, a list of
-// InformerSynced functions that must be synced, or an error.
-func (c *controller) Register(ctx *controllerpkg.Context) (workqueue.RateLimitingInterface, []cache.InformerSynced, error) {
-	// construct a new named logger to be reused throughout the controller
-	c.log = logf.FromContext(ctx.RootContext, ControllerName)
-
-	// create a queue used to queue up items to be processed
-	c.queue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(time.Second*5, time.Minute*30), ControllerName)
-
-	// obtain references to all the informers used by this controller
-	orderInformer := ctx.SharedInformerFactory.Acme().V1().Orders()
-	issuerInformer := ctx.SharedInformerFactory.Certmanager().V1().Issuers()
-	challengeInformer := ctx.SharedInformerFactory.Acme().V1().Challenges()
-	secretInformer := ctx.KubeSharedInformerFactory.Core().V1().Secrets()
-	// build a list of InformerSynced functions that will be returned by the Register method.
-	// the controller will only begin processing items once all of these informers have synced.
+// NewController constructs an orders controller using the provided options.
+func NewController(
+	log logr.Logger,
+	cmClient cmclient.Interface,
+	kubeInformerFactory informers.SharedInformerFactory,
+	cmInformerFactory cminformers.SharedInformerFactory,
+	accountRegistry accounts.Getter,
+	recorder record.EventRecorder,
+	clock clock.Clock,
+	isNamespaced bool,
+) (*controller, workqueue.RateLimitingInterface, []cache.InformerSynced) {
+
+	// Create a queue used to queue up Orders to be processed.
+	queue := workqueue.NewNamedRateLimitingQueue(
+		workqueue.NewItemExponentialFailureRateLimiter(time.Second*5, time.Minute*30),
+		ControllerName,
+	)
```

Comment:
Are these baseDelay and maxDelay defined anywhere else in cert-manager? It seems like all the controllers should use the same rate limiting parameters.

Reply:
It seems to be 1s,30s everywhere apart from orders controller and challenges informer where it's 5s,30min.

Agree that we should make them into constants and, if there is a specific reason why the orders and challenges have different `baseDelay` that should be documented. I've created an issue for this #4014